### ISSUE TITLE
It's too easy to lose a project by dragging another one in from the desktop

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap#2733]
This is from a teacher on Piazza, who says this happens to her students a bunch.

There's no way to overwrite a project accidentally with the Open dialog; it takes too many steps.  But dragging an XML file in from the local desktop is easy, and is even the right thing if what's in the XML is a sprite or a library.  And because of the experience with sprites, a user might naively expect that dragging a project onto another project will merge them.

So, how about a warning dialog box if a user drags a project file onto a nonempty project?

**Comment by jmoenig** [comment]
yeah, I think our plans for autosave to localstore is going to eventually take care of stuff like this. But I'm also slightly annoyed by such feedback. I mean, are they using Snap! offline? Then simply saving the project by hitting ctrl-s will export a new version of it to their computers. Are they using Snap! online? Then why are they even dragging XML projects in? There's a cloud for that. And if they - for unfathomable reasons - decide to mix both modes there's the ancient maxim of "save your work".

Anyway, I'm very much against any further nag boxes, because in my experience they are not going to solve any issues. Autosave will take care of it, but don't hold your breath.

**Comment by brianharvey** [comment]
They don't save often enough because they're children.  They drag projects in because they're children.  C'mon, don't shoot the messenger.  (The teacher, I mean; you can shoot me if you want.)

**Comment by jmoenig** [comment]
Sorry, but it sounds like a case of "the dog ate my homework".

**Comment by brianharvey** [comment]
It's not homework -- well, I guess this year it is, but generally they'd be working in the classroom.

Oh well I tried.

**Comment by jguille2** [comment]
Hi Jens,

But here, I see some sense in this issue.

In our cloud we can store projects... but if you want to play with custom libraries, or sprites, or scripts... then you are using "xml" files.
And yes, both dragging that file or importing it, maybe you think it is a library (or a sprite...) but it's a project and then, your current project is gone.

Then, I think the request is to add a "dialog" only when that file is a "<project>". Maybe it is not bad.

Joan
